drivers/soundwire/cadence_master.c

```diff
@@ -368,6 +372,97 @@ static int cdns_hw_reset(void *data, u64 value)
 
 DEFINE_DEBUGFS_ATTRIBUTE(cdns_hw_reset_fops, NULL, cdns_hw_reset, "%llu\n");
 
+static int cdns_parity_error_injection(void *data, u64 value)
+{
+	struct sdw_cdns *cdns = data;
+	struct sdw_slave *slave;
+	struct sdw_bus *bus;
+	int ret;
+
+	if (value != 1)
+		return -EINVAL;
+
+	bus = &cdns->bus;
+
+	/*
+	 * wake up master and slave(s)
+	 */
+	list_for_each_entry(slave, &bus->slaves, node) {
```

Comment:
This looks a bit crude to me, but I don't have a better solution either. I assume there isn't a single device that we could wake up that would wake up all the dependencies?

Reply:
I just found https://github.com/thesofproject/linux/pull/2247/commits/31527f485a392be1d0011b6ae6981d62e28940e6 - doesn't it mean, that the same can be done here?

Reply:
yes we can, and that would simplify the whole pm stuff.

Basically if we resume the master then all slaves will be re-enumerated and will be resumed.